### PR TITLE
data_structures/linked_list: Add __len__() function and tests

### PR DESCRIPTION
**Pull request by Lakshmikanth2001**
### **Describe your change:**

* [x] Add an algorithm

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/linked_list/__init__.py [hunk 1]
```diff
@@ -2,21 +2,24 @@ class Node:
...
             return None
         else:
             item = self.head.item
             self.head = self.head.next
+            self.size-=1
             return item
-
     def is_empty(self):
         return self.head is None
+    def __len__(self): #add python predefined len function
```
Comment (cclauss) [inline data_structures/linked_list/__init__.py:24]:
```suggestion
    def __len__(self):
```

File: maths/prime_check.py [hunk 2]
```diff
@@ -22,8 +22,9 @@ def prime_check(number):
 
     # Except 2, all primes are odd. If any odd value divide
     # the number, then that number is not prime.
-    odd_numbers = range(3, int(math.sqrt(number)) + 1, 2)
-    return not any(number % i == 0 for i in odd_numbers)
+    # prime number all are in 6k+1 or 6k-1 form where k belongs to integers so i have modified range(5,root(n),6)
```
Comment (cclauss) [inline maths/prime_check.py:25]: Please wrap all lines at 88 chars max.

File: maths/prime_check.py [hunk 3]
```diff
@@ -22,8 +22,9 @@ def prime_check(number):
...
+    special_number=range(5,int(math.sqrt(number)),6)
```
Comment (cclauss) [inline maths/prime_check.py:26]: I find the name `special_number` to be confusing.  Can we have a more self documenting variable name?  What exactly is _special_ about this number?

File: data_structures/linked_list/__init__.py [hunk 4]
```diff
@@ -2,21 +2,24 @@ class Node:
     def __init__(self, item, next):
         self.item = item
         self.next = next
-
 class LinkedList:
     def __init__(self):
         self.head = None
+        self.size=0
```
Comment (cclauss) [inline data_structures/linked_list/__init__.py:8]:
```suggestion
        self.size = 0
```
psf/black is just going to undo all this text with no room to breathe.

File: maths/prime_check.py (hunk 3 above)
Comment (Lakshmikanth2001) [reply maths/prime_check.py:26]: Sir here special number are number in the form 6k+1 or 6k-1 every prime number will be in the form of 6k+1 or 6k-1

**Pull request merged**